tools/ci_build/build.py

```diff
@@ -1472,8 +1472,12 @@ def generate_build_tree(
         ):
             if is_windows():
                 cflags = ["/guard:cf", "/DWIN32", "/D_WINDOWS"]
-                if args.parallel:
-                    cflags += ["/MP"]
+                njobs = number_of_parallel_jobs(args)
+                if njobs > 1:
+                    if args.parallel == 0:
+                        cflags += ["/MP"]
+                    else:
+                        cflags += ["/MP%d" % njobs]
```

Comment:
actually I wonder if this should just be done in build_targets(). it looks like "CL_MPCount" is already passed there.

https://github.com/microsoft/onnxruntime/blob/28a16c223cb7a8707fb90340c8aecdc46eec3f2f/tools/ci_build/build.py#L1618-L1625

though should we also set the "EnforceProcessCountAcrossBuilds" and "UseMultiToolTask" properties?

https://devblogs.microsoft.com/cppblog/improved-parallelism-in-msbuild/

Reply:
The code you mentioned do not get executed when building the code outside of cmake. For example, a lot of people use "--update" to generate the sln file then directly open the file in Visual Studio and build it there. Also, most our CI build pipelines do the same.